TypeAdapter generator

gonzodepedro opened the pull request:
Adds automatic rclcpp::TypeAdapter generation for protobuf.

**Note:** Code is ready for review, documentation will come later.
**Note:** To be merged after #26

gonzodepedro committed:
Added rosil_generator_cpp dependency to rosidl_typesupport_protobuf_c

Signed-off-by: Gonzo de Pedro <gonzo@depedro.com.ar>

gonzodepedro committed:
Added new template for type adaptation. plus other fixes

Signed-off-by: Gonzo de Pedro <gonzo@depedro.com.ar>

gonzodepedro committed:
changes to include missing dependencies

gonzodepedro committed:
Added ament_export_typesupport_targets and making it work

Signed-off-by: Gonzo de Pedro <gonzo@depedro.com.ar>

gonzodepedro committed:
Multiple changes

Signed-off-by: Gonzo de Pedro <gonzo@depedro.com.ar>

gonzodepedro committed:
Changes in the dependency files

Signed-off-by: Gonzo de Pedro <gonzo@depedro.com.ar>

gonzodepedro committed:
Changed test names. Fixed some linter

Signed-off-by: Gonzo de Pedro <gonzo@depedro.com.ar>

gonzodepedro committed:
Fixed wstring test

Signed-off-by: Gonzo de Pedro <gonzo@depedro.com.ar>

gonzodepedro committed:
Fixed linter

Signed-off-by: Gonzo de Pedro <gonzo@depedro.com.ar>

gonzodepedro committed:
Fix cmakelist lint

Signed-off-by: Gonzo de Pedro <gonzo@depedro.com.ar>

gonzodepedro committed:
Added missing dependencies

Signed-off-by: Gonzo de Pedro <gonzo@depedro.com.ar>

gonzodepedro committed:
Fixed dependencies. Fixed fucntion name colision

Signed-off-by: Gonzo de Pedro <gonzo@depedro.com.ar>

gonzodepedro committed:
Fixed linters

Signed-off-by: Gonzalo de Pedro <gonzalodepedro@ekumenlabs.com>

FlorianReimold commented on rosidl_typesupport_protobuf_cpp/resource/idl__typeadapter_protobuf_cpp.hpp.em:
```diff
@@ -0,0 +1,135 @@
+)
+}@
+
+<<<<<<< HEAD
```
A wild merge conflict appears 🐛

gonzodepedro replied:
Fixed

FlorianReimold commented:
Hm, GitHub now prompts me to resolve some merge conflicts between your fork and the master branch

tfoote committed:
add new package to CI packages to cover

tfoote committed:
disable linters for the moment due to Continental copyright notice failure to match
It's a false positive, but ament_copyright is moderately hard to extend.
This should get it passing for the moment.

tfoote commented:
I rebased the changes onto the recent PRs. It's passing CI now. 

The copyright tests were failing due to being overly strict. I've ticketed to restore them at https://github.com/tfoote/rosidl_typesupport_protobuf/issues/11

gonzodepedro commented:
**Attention: this branch has changed and been force pushed**

gdiaz-guevara committed:
Protobuf document (#9)

* Add a README.md for the typeadapter_protobuf

FlorianReimold commented:
I am currently checking back with Eclipse if there are any processes to follow, as this contribution is very big and reviewing it line by line is probably not the best approach here.

gonzodepedro commented:
We will try to split this into multiple shorter and hopefully easier to review PRs.

gonzodepedro commented:
We have split this PR into 3 more manageable PRs.
Closing this, in favor of #30, #31 and #32
